src/backend/gporca/libgpopt/src/translate/CTranslatorExprToDXL.cpp

```diff
@@ -449,6 +449,22 @@ CTranslatorExprToDXL::CreateDXLNode(CExpression *pexpr,
 
 		return dxlnode;
 	}
+	if (COperator::EopPhysicalMotionGather == ulOpId ||
+		COperator::EopPhysicalMotionBroadcast == ulOpId ||
+		COperator::EopPhysicalMotionHashDistribute == ulOpId ||
+		COperator::EopPhysicalMotionRoutedDistribute == ulOpId ||
+		COperator::EopPhysicalMotionRandom == ulOpId)
+	{
+		gpos::IntPtrArray *inputSegmentInfo = GetInputSegIdsArray(pexpr);
+
+		if (NULL != inputSegmentInfo)
```

Comment:
Function GetInputSegIdsArray cannot return NULL, so this check is redundant

Reply:
Done